Content.Shared/ImmovableRod/ImmovableRodSystem.cs

```diff
@@ -0,0 +1,161 @@
+        _physics.SetBodyStatus(uid, phys, BodyStatus.InAir);
+
+        var xform = Transform(uid);
+        var (worldPos, worldRot) = _transform.GetWorldPositionRotation(uid);
+        var vel = worldRot.ToWorldVec() * component.MaxSpeed;
+
+        if (component.RandomizeVelocity)
+        {
+            vel = component.DirectionOverride.Degrees switch
+            {
+                0f => _random.NextVector2(component.MinSpeed, component.MaxSpeed),
+                _ => worldRot.RotateVec(component.DirectionOverride.ToVec()) * _random.NextFloat(component.MinSpeed, component.MaxSpeed)
+            };
+        }
+
+        _physics.ApplyLinearImpulse(uid, vel, body: phys);
+        xform.LocalRotation = (vel - worldPos).ToWorldAngle() + MathHelper.PiOver2;
+    }
+
+    protected virtual void OnCollide(EntityUid uid, ImmovableRodComponent component, ref StartCollideEvent args)
+    {
+        var ent = args.OtherEntity;
+
+        // Clang!
+        if (_random.Prob(component.HitSoundProbability))
+            _audio.PlayPvs(component.Sound, uid);
+
+        // Oh nyo.
+        if (HasComp<ImmovableRodComponent>(ent))
+            HandleRodRodCollision(uid, ent);
+        // Mobs never automatically get deleted by Rodney, because being instantly deleted is not any fun.
+        else if (TryComp<MobStateComponent>(ent, out var mob))
+            HandleRodMobCollision(uid, component, args);
+        // Admiral Roddington XIV may have been limited in his desire to delete everythng.
+        // What's the point of an immovable rod if it deletes all those organs it's just splattered?
+        else if (component.ShouldDelete && !_whitelist.IsWhitelistPass(component.DeletionBlacklist, ent))
+            QueueDel(ent);
+        // The Amulet of Yendor has powers some may deem... unnatural. Clang!
+        else if (component.Damage != null)
+            _damageable.TryChangeDamage(ent, component.Damage, ignoreResistances: true);
+    }
+
+    private void HandleRodRodCollision(EntityUid uid, EntityUid ent)
+    {
+        // :panik:
+        var coords = Transform(uid).Coordinates;
+        _popup.PopupCoordinates(Loc.GetString("immovable-rod-collided-rod-not-good"), coords, PopupType.LargeCaution);
+
+        // :kalm:
+        Del(uid);
+        Del(ent);
+
+        // :panik:
+        Spawn("Singularity", coords);
+    }
+
+    private void HandleRodMobCollision(EntityUid uid, ImmovableRodComponent component, StartCollideEvent args)
+    {
+        var ent = args.OtherEntity;
+
+        // TODO: Immovable rods are weird because they're projectiles that don't get consumed after they collide.
+        // We could store who the projectile has already hit, but prediction makes that non-trivial, and I've
+        // not found a good example in the codebase that mimics the rod's desired behaviour.
+        // Temporary hack: use the same collision layer that MobCollision uses.
+        if (args.OtherFixtureId != "flammable")
```

Comment:
perhaps just check for a hard fixture instead of specifically checking this one. Lots of entities use non-hard fixtures like this for checking special types of collision.

Reply:
We need something consistent and sane for "thing hit mob". I'm deferring to "mirror how mob collision works"; we need a more universal system. Like a lot with mob code it's just shitcode under the hood.